templates/set-absolute-and-idle-timeout-based-on-organization-connection-or-application-metadata-POST_LOGIN/code.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Handler that will be called during the execution of a PostLogin flow.
+ *
+ * @param {Event} event - Details about the user and the context in which they are logging in.
+ * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+ */
+exports.onExecutePostLogin = async (event, api) => {
+    let currentTime = new Date().getTime();
+  
+    // Get organization or client metadata (organization takes precedence if present)
+    const metadata = event.organization?.metadata || event.client?.metadata || event.connection?.metadata;
+  
+    if (metadata) {
+      // Set idle timeout
+      const idleTimeout = Number(metadata.idle_lifetime);
```

Comment:
I think it would be worth considering lifting the `idle_lifetime` and `absolute_lifetime` fields to top-level constants and documenting them so that customers can easily pick other metadata field names for these conventions.

Reply:
@ggoodman 

Do you mean something like

`
**
 * Constants for session metadata field names.
 * In you can create those metadata values in Organizations, Connections or inside Applications in Auth0. If the metadata is created, this action will use it
 */
const IDLE_LIFETIME_FIELD = 'idle_lifetime';
const ABSOLUTE_LIFETIME_FIELD = 'absolute_lifetime';

`

Then

` const idleTimeout = Number(metadata[IDLE_LIFETIME_FIELD]);`

?

Reply:
@nelsonmaia, yes exactly. I was thinking that might give us an opportunity to turn those magic string constants into self-documenting constants. I think this might help us reduce reliance on separate docs and to clear up ambiguity.

For example, I don't know if these constants are well-known values that are platform-defined or whether they're more of a convention. The comments on the constants can help demystify this for users.